### PR TITLE
Feature: Authentication

### PR DESCRIPTION
**Pull request by panlabdev**
Implement simple authentication using:

- Credentials
- Facebook
- Google

Override API endpoint for signup.

Create lib for auth, prisma, and validation schemas.

Fix small schema problem.

Implement simple authorization without role.

New Dependencies:

- bcrypt
- zod

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -41,17 +41,18 @@ model Session {
...
+  email          String?           @unique
+  emailVerified  DateTime?
+  hashedPassword String?
+  image          String?
+  phone          String?
+  role           Role              @default(CUSTOMER)
+  accounts       Account[]
+  sessions       Session[]
+  orders         Order[]
+  addresses      CustomerAddress[]
```
Comment (tarlrsk) [inline prisma/schema.prisma:55]: nullable ?

File: .gitignore [hunk 2]
```diff
@@ -25,13 +25,14 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 
-# local env files
+# env files
 .env*.local
-.env
+.env*
```
Comment (tarlrsk) [inline .gitignore:31]: แค่ .env*  ก็พอป้ะ

**Review by tarlrsk — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (panlabdev) [reply .gitignore:31]: Cannot for some reason. It still .env.development still appears in the repo after adding .env*

Comment (tarlrsk) [reply .gitignore:31]: แต่กุได้นะมึง

File: src/app/api/auth/signup/route.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import { responseWrapper } from "@/utils/api-response-wrapper";
+import { NextRequest } from "next/server";
+
+export async function POST(req: NextRequest) {
+  try {
+    const { name, email, password } = await req.json();
+
+    if (name === null || name === "") {
+      return responseWrapper(400, null, "Name is required.");
+    }
```
Comment (tarlrsk) [inline src/app/api/auth/signup/route.ts:15]: if (!name?.trim()) {
  return responseWrapper(400, null, "Name is required.");
} else if (!email?.trim()) {
  return responseWrapper(400, null, "Email is required.");
} else if (!password?.trim()) {
  return responseWrapper(400, null, "Password is required.");
}

**Review by tarlrsk — APPROVED** [review]

**Pull request merged**